Add initial iOS template

gfontenot opened the pull request:
This template is a slightly modified version of the one generated by Liftoff.
It represents the basics for how we'd want to set a project up. I didn't add
or change _too_ much because I wanted to make any semi-controversial changes
in individual PRs.

gfontenot committed:
Add initial iOS template

This template is a slightly modified version of the one generated by Liftoff.
It represents the basics for how we'd want to set a project up. I didn't add
or change _too_ much because I wanted to make any semi-controversial changes
in individual PRs.

gfontenot commented:
The idea here is that we'd have a tool that is able to pull this template down and replace the placeholder text with the company name and project name and whatnot and then we'd have a project ready to go. Basically: a much simpler and not-at-all configurable version of Liftoff. This template would be our starting point for projects moving forward.

tonyd256 commented on Sources/Controllers/AppDelegate.swift:
```diff
@@ -0,0 +1,6 @@
+import UIKit
+
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate {
```
what about final :)

gfontenot replied:
Oh yep

tonyd256 commented on Sources/Resources/Other-Sources/Info.plist:
```diff
@@ -0,0 +1,40 @@
+	<key>CFBundlePackageType</key>
+	<string>APPL</string>
+	<key>CFBundleShortVersionString</key>
+	<string>1.0</string>
```
what about making these versions 1.0.0 to work better with FastLane, assuming we want to use that more

gfontenot replied:
👍

tonyd256 commented on bin/bootstrap:
```diff
@@ -0,0 +1,6 @@
+
+set -e
+
+carthage bootstrap --platform iOS --no-use-binaries
```
are we back to having carthage build for us instead of adding the projects to Xcode?

gfontenot replied:
AFAIK, we've been doing it this way for a _really_ long time now. All of my most recent projects have used the binaries created by Carthage. Is there something I don't know?

gfontenot committed:
Use proper semver

gfontenot committed:
Mark classes as `final`

gfontenot committed:
Fix indentation

gfontenot committed:
Use the master branch for Argo

The newest version of Carthage (0.18) fixes the issue that I was trying
to fix in Argo, so we no longer need to use a special branch to build
properly.

gfontenot committed:
Update circle.yml for Carthage config

gfontenot commented:
CI is broken because Circle doesn't have the newest beta up yet (I... I think)

jakecraige commented on Gemfile:
```diff
@@ -0,0 +1,3 @@
+source 'https://rubygems.org'
+
+gem 'xcpretty'
```
This is ridiculously nitpicky, but on the ruby side, our style guide prefers double quotes as the default

gfontenot replied:
¯\_(ツ)_/¯ sure

jakecraige commented:
Works for me 👍

gfontenot committed:
Use double quotes for Ruby

This brings us in line with the thoughtbot Ruby styleguide.

gfontenot committed:
Properly link Runes as a dependency

Without this our app crashes at launch. Runes is a dependency of Argo
and so needs to be linked explicitly.

gfontenot committed:
Upgrade Carthage before installing dependencies

gfontenot commented:
We're back to the CI failures being out of our control. Once they update to the Xcode GM it should work again.

gfontenot committed:
Use more unique variable names

It's reasonable to think that using such generic names could result in
false positives during the find/replace process. For example, it seems
completely likely that replacing `BUNDLE_IDENTIFIER` would result in
messing up the `PRODUCT_BUNDLE_IDENTIFIER` key.

Using a prefix of `KICKOFF` for these keys should reduce the likelyhood
of this kind of thing happening.
